nighthawk/test/server/http_test_server_filter_integration_test.cc

```diff
@@ -0,0 +1,262 @@
+  EXPECT_STREQ("bar2",
+               response->headers().get(Envoy::Http::LowerCaseString("foo"))->value().c_str());
+  EXPECT_EQ(std::string(10, 'a'), response->body());
+}
+
+class HttpTestServerIntegrationNoConfigTest : public HttpTestServerIntegrationTestBase {
+public:
+  void SetUp() override { initialize(); }
+
+  void TearDown() override {
+    cleanupUpstreamAndDownstream();
+    test_server_.reset();
+    fake_upstreams_.clear();
+  }
+
+  void initialize() override {
+    config_helper_.addFilter(R"EOF(
+name: test-server
+)EOF");
+    HttpTestServerIntegrationTestBase::initialize();
+  }
+};
+
+INSTANTIATE_TEST_CASE_P(IpVersions, HttpTestServerIntegrationNoConfigTest,
+                        testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()));
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestNoHeaderConfig) {
+  Envoy::BufferingStreamDecoderPtr response =
+      makeSingleRequest(lookupPort("http"), "GET", "/", "", downstream_protocol_, version_,
+                        "foo.com", "", [](Envoy::Http::HeaderMapImpl&) {});
+  ASSERT_TRUE(response->complete());
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+  EXPECT_EQ(std::string(0, 'a'), response->body());
+}
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestBasics) {
+  testWithResponseSize(1, false);
+  testWithResponseSize(10, false);
+  testWithResponseSize(100, false);
+  testWithResponseSize(1000, false);
+  testWithResponseSize(10000, false);
+}
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestNegative) { testBadResponseSize(-1); }
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestZeroLengthRequest) {
+  testWithResponseSize(0, false);
+}
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestMaxBoundaryLengthRequest) {
+  const int max = 1024 * 1024 * 4;
+  testWithResponseSize(max, false);
+}
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestTooLarge) {
+  const int max = 1024 * 1024 * 4;
+  testBadResponseSize(max + 1);
+}
+
+TEST_P(HttpTestServerIntegrationNoConfigTest, TestHeaderConfig) {
+  Envoy::BufferingStreamDecoderPtr response = makeSingleRequest(
+      lookupPort("http"), "GET", "/", "", downstream_protocol_, version_, "foo.com", "",
+      [](Envoy::Http::HeaderMapImpl& request_headers) {
+        const std::string header_config =
+            "{response_headers: [ { header: { key: \"foo\", value: \"bar2\"}, append: true } ]}";
+        request_headers.addCopy(Nighthawk::Server::TestServer::HeaderNames::get().TestServerConfig,
+                                header_config);
+      });
+  ASSERT_TRUE(response->complete());
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+  EXPECT_STREQ("bar2",
+               response->headers().get(Envoy::Http::LowerCaseString("foo"))->value().c_str());
+  EXPECT_EQ(std::string(0, 'a'), response->body());
+}
+
+// TODO(oschaaf): maybe add a couple more tests for appending headers.
```

Comment:
If the header merge/append logic is factored out into its own module, we can just do unit tests as well.

Reply:
Sure; would you like to see that in this PR? Or would you be ok with doing that in a follow up?

Reply:
I reckon let's do it here since it's probably not a ton of code, but up to you as well.